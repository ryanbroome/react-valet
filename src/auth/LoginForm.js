import React, { useState, useContext } from "react";
import { useHistory } from "react-router-dom";
import { FormGroup, Form, Input, Label, Button } from "reactstrap";
import UserContext from "./UserContext";

const LoginForm = () => {
  const history = useHistory();
  const { login } = useContext(UserContext);

  const INITIAL_STATE = {
    username: "",
    password: "",
  };

  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((formData) => ({
      ...formData,
      [name]: value,
    }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    login(formData.username, formData.password);
    setFormData(INITIAL_STATE);
    history.push(`/`);
  };

  return (
    <Form onSubmit={handleSubmit}>
      <h3>Login</h3>
      {Object.keys(INITIAL_STATE).map((val, idx) => (
        <FormGroup key={`FormGroup-${val}`}>
          <Label
            htmlFor={val}
            key={`Label-${idx}`}></Label>
          <Input
            id={val}
            key={`Input-${val}`}
            type="text"
            name={val}
            placeholder={val.toLowerCase()}
            value={formData[val]}
            onChange={handleChange}></Input>
        </FormGroup>
      ))}

      <Button color="primary">Login</Button>
    </Form>
  );
};

export default LoginForm;
