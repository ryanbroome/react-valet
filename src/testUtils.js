import React from "react";
import UserContext from "./auth/UserContext";
// import { render } from "@testing-library/react";
// import { MemoryRouter } from "react-router";
import App from "../src/App";
// import TransactionList from "./TransactionList";

const MockUserContext = ({ children }) => {
  const mockToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InRlc3R1c2VyIiwibG9jYXRpb25JZCI6MSwiaXNBZG1pbiI6ZmFsc2UsImlhdCI6MTcxMzA3NTM4Mn0.dXlnawfzk6wlsgpouoSp58QLvlOOSJFC03iq_FXab-I";
  const mockUserDetail = {
    userId: 1,
    vehicleId: 1,
    locationId: 1,
    transactionId: 1,
    ticketNum: "1",
    mobile: "5553334444",
    color: "red",
    make: "honda",
    username: "testuser",
  };
  const mockLocation = {};
  const mockVehicle = {};
  const mockTransaction = {};

  return <UserContext.Provider value={{ token: mockToken, userDetail: mockUserDetail, location: mockLocation, vehicle: mockVehicle, transaction: mockTransaction, update: App.update }}>{children}</UserContext.Provider>;
};

export { MockUserContext };
