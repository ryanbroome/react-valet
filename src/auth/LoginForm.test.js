import React from "react";
import { render } from "@testing-library/react";
import LoginForm from "./LoginForm";
import { MemoryRouter } from "react-router";
import { MockUserContext } from "../testUtils";

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <MockUserContext children={<LoginForm />}></MockUserContext>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
