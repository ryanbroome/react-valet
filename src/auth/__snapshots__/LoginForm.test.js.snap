// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <h3>
      Login
    </h3>
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="username"
      />
      <input
        class="form-control"
        id="username"
        name="username"
        placeholder="username"
        type="text"
        value=""
      />
    </div>
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="password"
      />
      <input
        class="form-control"
        id="password"
        name="password"
        placeholder="password"
        type="text"
        value=""
      />
    </div>
    <button
      class="btn btn-primary"
    >
      Login
    </button>
  </form>
</DocumentFragment>
`;
